app/controllers/admin/bookings_controller.rb

```diff
@@ -28,7 +28,30 @@ def load_booking
     redirect_to admin_bookings_path
   end
 
+  def search_bookings
+    @search_params = search_booking_params
+    start_at = format_date @search_params[:start_at]
+    created_at = format_date @search_params[:created_at]
+    @bookings = if @search_params.blank?
+                  Booking.newest
+                else
+                  Booking
+                    .search_by_booking_status(@search_params[:booking_status])
+                    .search_by_start_at(start_at)
+                    .search_by_created_at(created_at)
+                    .newest
+                end
+  end
+
+  def format_date string
+    Date.parse(string) if string.present?
+  end
+
   def booking_params
-    params.require(:booking).permit(:date, :booking_status, :football_pitch_id)
+    params.require(:booking).permit(:booking_status)
+  end
+
+  def search_booking_params
+    params.permit(:start_at, :booking_status, :created_at)
```

Comment:
gọi thẳng params luôn k cần tạo hàm để permit đâu. chỉ tạo / cập nhập ms cần. còn search thì k cần đâu e

Reply:
toẹt vời, kiến thức này đã được tiếp thu